cubical/solver.py

```diff
@@ -85,6 +85,9 @@ def _solve_gains(gm, stats, madmax, obser_arr, model_arr, flags_arr, sol_opts, l
 
     diverging = ""
 
+    # for cases the are no valid solutions
+    gm.output_weights = None 
```

Comment:
Is this safe?

Reply:
Yes this and the one below safe. i.e. self.output_weights = self.gm.output_weights
Actually initializing the weights to None is for the complex solver and all the other solvers which do not output weights.